### PR TITLE
refactor(tests): Simplify test logic, add nesting, display annotations, and refine variable names and added few tests

### PR DESCRIPTION
**Pull request by balazs-szucs**
# Description of Changes
This PR refactors multiple test classes (e.g., ConvertWebsiteToPdfTest, UploadLimitServiceTest, PdfMetadataServiceTest, SignatureServiceTest) with non-aggressive improvements to enhance code quality, and readability:
- Added `@Nested` and `@DisplayName` annotations for better test organization and grouping of logically similar tests.
- Added names to `@ParameterizedTest` annotations for clearer test descriptions.
- Removed unused imports to clean up the code.
- Improved null checks and added edge cases for more robust testing (e.g., null/empty inputs).
- Simplified assertion logic (e.g., replaced `assertTrue(X instanceof Y)` with `assertInstanceOf`).
- Enhanced variable, method, and test value names for better clarity.
- Improved comments for better documentation.
- Added `assertNotNull` checks and requireNonNull in relevant case
- Refined several test cases that were lackluster, including better handling of exceptions and inputs.

In the grand scheme of things, there are very few large/meaningful changes rather, many small quality-of-life changes that might help avoid some headaches down the line in the future (hopefully).

---

### General

- [x] I have read the [Contribution Guidelines](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/CONTRIBUTING.md)
- [x] I have read the [Stirling-PDF Developer Guide](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/devGuide/DeveloperGuide.md) (if applicable)
- [ ] I have read the [How to add new languages to Stirling-PDF](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/devGuide/HowToAddNewLanguage.md) (if applicable)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

### Documentation

- [ ] I have updated relevant docs on [Stirling-PDF's doc repo](https://github.com/Stirling-Tools/Stirling-Tools.github.io/blob/main/docs/) (if functionality has heavily changed)
- [ ] I have read the section [Add New Translation Tags](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/devGuide/HowToAddNewLanguage.md#add-new-translation-tags) (for new translation tags only)

### UI Changes (if applicable)

- [ ] Screenshots or videos demonstrating the UI changes are attached (e.g., as comments or direct attachments in the PR)

### Testing (if applicable)

- [x] I have tested my changes locally. Refer to the [Testing Guide](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/devGuide/DeveloperGuide.md#6-testing) for more details.

### CODE REVIEW

File: app/core/src/test/java/stirling/software/SPDF/controller/web/UploadLimitServiceTest.java [hunk 1]
```diff
@@ -56,13 +59,14 @@ static Stream<Arguments> uploadLimitParams() {
                 // valid formats
                 Arguments.of("10KB", 10 * 1024L),
                 Arguments.of("2MB", 2 * 1024 * 1024L),
-                Arguments.of("1GB", 1L * 1024 * 1024 * 1024),
+                Arguments.of("1GB", (long) 1024 * 1024 * 1024),
```
[the remark this answers is not in the document]
Comment (balazs-szucs) [reply app/core/src/test/java/stirling/software/SPDF/controller/web/UploadLimitServiceTest.java:62]: In my opinion first better for readability. But open for change if second (co pilot suggested) is more aligned with the codebase.

**Comment by balazs-szucs** [comment]
Hi,

Sonarqube seem to be failing, however I am not too sure if I did something wrong or not. For instance: EmailControllerTest is on proprietary module so I obviously didn't really touch that and yet it errored. Also at first this PR went through everything fine, and now this so that's interesting. Nonetheless if it's something on my side please feel free to let me know. :)

EDIT: it was fixed by someone in the meantime, was def unrelated to this PR.

**Comment by balazs-szucs** [comment]
Hi,

After long consideration, I decided that this PR should be closed. While in my (personal) opinion the changes are worthwhile, I realized:
- It is not fair nor inline with best practices to submit such large PR
- Even though I had best intentions in mind, I realize this PR might not perceived, due to several factors (like the already mentioned size)
- This would be huge work, and burden on me (which I don't mind) but more importantly on the maintainers to follow/review etc... so many changes scattered across so many files

Please feel free to add feedback or opinions. I can obviously re-open if that's some you want or reintroduce the changes in smaller PRs.
